app/controllers/budgets_controller.rb

```diff
@@ -6,13 +6,12 @@ class BudgetsController < ApplicationController
 
   def index
     @budgets = policy_scope(current_user.budgets.includes(:category))
-    # TODO: Refactor this code, move budget spending calucation to budget model
-    @spent_amounts = @budgets.map { |budget| calculate_spent(budget) }
+    @spent_amounts = @budgets.map(&:spent_amount)
```

Comment:
No need to create a separate instance variable, you can just call `budget.spent_amount` in views where you need to display this information.